smart-contracts/assembly/contracts/website-deployer/websiteStorer.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Website storer implementation by MassaLabs.
+ *
+ * This smartcontract is used to upload and manage website chunks.
+ *
+ */
+
+import {
+  Storage,
+  Context,
+  generateEvent,
+  callerHasWriteAccess,
+} from '@massalabs/massa-as-sdk';
+import {
+  Args,
+  bytesToI32,
+  i32ToBytes,
+  stringToBytes,
+} from '@massalabs/as-types';
+import { onlyOwner } from '../utils';
+import { _setOwner } from '../utils/ownership-internal';
+
+export const NB_CHUNKS_KEY = stringToBytes('NB_CHUNKS');
+
+export function constructor(_: StaticArray<u8>): void {
+  assert(callerHasWriteAccess());
+  _setOwner(Context.caller().toString());
+}
+
+/**
+ * This function is doing the following:
+ *    - Add a chunk to the newly created key chunkNb.
+ * 		- Block Upload if not dnsName owner
+ *
+ * @param binaryArgs - chunk id (u64), content (uint8Array)
+ *
+ */
+export function appendBytesToWebsite(binaryArgs: StaticArray<u8>): void {
+  onlyOwner();
+
+  const args = new Args(binaryArgs);
+  const chunkNb = args.nextI32().expect('chunk number is missing or invalid');
+  const chunkData = args.nextBytes().expect('chunkData is missing or invalid');
+
+  Storage.set(i32ToBytes(chunkNb), chunkData);
+
+  const totalChunks: i32 = Storage.has(NB_CHUNKS_KEY)
+    ? bytesToI32(Storage.get(NB_CHUNKS_KEY))
+    : 0;
+  if (chunkNb >= totalChunks) {
+    Storage.set(NB_CHUNKS_KEY, i32ToBytes(chunkNb + 1));
+  }
+
+  generateEvent(
+    `Website chunk ${chunkNb} deployed to ${Context.callee().toString()}`,
+  );
+}
+
+export function deleteWebsite(_: StaticArray<u8>): void {
+  onlyOwner();
+
+  assert(Storage.has(NB_CHUNKS_KEY), 'Website not uploaded yet');
+  const nbChunks = bytesToI32(Storage.get(NB_CHUNKS_KEY));
+
+  for (let i: i32 = 0; i < nbChunks; i++) {
+    const key = i32ToBytes(i);
+    if (Storage.has(key)) {
+      Storage.del(key);
+    }
+  }
+  Storage.del(NB_CHUNKS_KEY);
+
```

Comment:
Why not also removing the owner record ?

Reply:
The data is deleted but the smartContract is potentially still linked to a dns entry, and the owner can still upload a new website using this storer SC